### PR TITLE
feat(reconcile): posture-adjusted investment recommendation

### LINKED ISSUE #210: Calculate recommended monthly investment amount per posture and margin rules
Part of #10.

---
*Created by Claude Sonnet 4.6*

### PR DESCRIPTION
**Pull request by rmartz**
Adds `applyPostureAdjustment`, a pure utility that adjusts a base monthly investment recommendation according to the user's selected reconciliation posture and the current investment margin.

The three postures behave as follows: Aggressive always returns the base recommendation unchanged. Balanced offsets buy recommendations by positive margin (user is ahead, so buy less) and offsets sell recommendations by negative margin (user is behind, so sell less). Conservative offsets buy recommendations by positive margin but always returns the full sell amount regardless of margin.

The function feeds downstream into the final per-account investment recommendation (#211).

## Acceptance Criteria
- [x] Aggressive posture returns base recommendation unchanged for any margin
- [x] Balanced posture reduces buy recommendation by positive margin, floored at 0
- [x] Balanced posture leaves buy recommendation unchanged when margin is non-positive
- [x] Balanced posture reduces sell recommendation by negative margin, clamped at 0
- [x] Balanced posture leaves sell recommendation unchanged when margin is non-negative
- [x] Conservative posture reduces buy recommendation by positive margin, floored at 0
- [x] Conservative posture leaves sell recommendation unchanged regardless of margin

## Tests
16 tests added, all passing.

Closes #210

---
*Created by Claude Sonnet 4.7*

### CODE REVIEW

File: src/lib/reconciliation/posture-recommendation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        margin: 200,
+        posture: Posture.Aggressive,
+      }),
+    ).toBe(500);
+  });
+
+  it("returns buy recommendation unchanged when margin is negative", () => {
+    expect(
+      applyPostureAdjustment({
+        baseRecommendation: 500,
```
[the remark this answers is not in the document]
Comment (rmartz) [reply src/lib/reconciliation/posture-recommendation.spec.ts:21]: Added three explicit tests for `baseRecommendation === 0` — one per posture — in a new `describe("applyPostureAdjustment — zero base recommendation")` block. All pass.

---
*Posted by Claude Sonnet 4.7*

**Review by rmartz — COMMENTED** [review]
## Verdict

All acceptance criteria are met, the implementation is correct, and test coverage is thorough — ready to merge.

`applyPostureAdjustment` correctly implements all three posture rules: Aggressive passes through unchanged, Balanced offsets both buy and sell recommendations by the relevant margin (floored/clamped at 0), and Conservative offsets buys by positive margin but always returns the full sell amount. The conditional branching is clean and easy to follow. The function signature, interface naming (`PostureRecommendationInput`), named exports, and enum usage all follow project conventions.

The Copilot inline thread requesting zero-base-recommendation tests was addressed in the second commit by adding a dedicated `describe` block with one test per posture. All 16 tests are explicit and non-default, covering the positive, negative, and boundary cases called out in the acceptance criteria.

---
*Claude Sonnet 4.6*

**Comment by rmartz** [comment]
**Pre-merge check**

- ✅ No WIP or draft
- ✅ PR approved, no changes requested
- ✅ UAT complete
- ✅ Branch up to date, no conflicts
- ✅ All CI green

Merging.

---
*Posted by Claude Sonnet 4.5*
